### DISCUSSION TITLE
Purpose of notifications in Kimai

### DISCUSSION THREAD

**Discussion by bigvictorio** [kimai/kimai#4168] [General]
Hi,

I was wondering what purpose notifications serve in Kimai app. I could not find any information about it in the documentation, GitHub discussions, Kimai-webpage, etc. I'm talking about notifications in a web app under preferences > profile.
![image](https://github.com/kimai/kimai/assets/70698524/f8f0dde8-8f9b-496c-84f6-120f2047e1ba)
After enabling notifications, I didn't receive any notifications from anything other than "Thank you for agreeing to notifications!"

Is there someone who could provide me with more information? 
(Is this just basic preparation for notifications (like just integration)? So I could build a plugin with notifications? Or are there already some existing notifications that could be useful, but I just don't know about them? Or I just didn't discover them yet :D )

Thanks for the help :)
Appreciate it

**Comment by kevinpapst** [comment]
> Is this just basic preparation for notifications

This 👍  
One step after another...  

> So I could build a plugin with notifications

Yes

>  just don't know about them

There are none yet, but I have some ideas. Please also share your ideas about possible future notification!

**Comment by bigvictorio** [comment]
Hello Kevin,

Thanks for the reply,

Especially for the "timesheet section," I don't have any ideas. However, I have some ideas for the tasks plugin:
Notifications:
- if a new task is created
- Task assignment. Notification if you are assigned to a task (new task, existing task). And maybe extend this functionality to teams and team leaders. 
- If a task has been completed
- If a task has been Planned (and maybe some notification in a manner "task example will start in x days")
- If a task has been modified (edited), e.g., time budget, new comment, extending time (maybe notifications for team leaders, and if they are not team leaders - (somebody is not using teams feature in kimai), notify the assigned user of the task.)
- Change of activity in the task
- Mentions (Mentions probably don't even exist in kimai), but such a feature would be helpful. Let's say I will open a task -> comments section, and in there, I could mention someone like "@bigvictorio opinion on this situation?" or "@JohnDee@example.com fix this ASAP!!!!". It should notify somebody you mentioned in that task etc. (beneficial for an update on some situations/projects etc.))
- When the deadline of the task approaches, 
- When the task is after the deadline (for some people, the red color of that task is not enough :D)
- Task is in progress (you started processing the selected task)
- At the End of the month, notify users to check all recorded times/tasks and expenses (I know there are already plugins for this - **HasWorked**), but maybe some functionality from this plugin would be helpful
- If the task deadline has been changed (period change - shortened to or extended to)

These are just ideas, maybe some of them are bad, and I hope some of them are not that bad.

Thanks

**Comment by kMikaZu** [comment]
IS there any news on this matter? It really would be handy to get notifications for Tasks (via mail or a real Push Notification).
